test/evp_extra_test2.c

```diff
@@ -1326,6 +1326,24 @@ static int test_evp_pbe_alg_add(void)
 }
 #endif
 
+/*
+ * Currently, EVP_<OBJ>_fetch doesn't support
+ * colon separated alternative names for lookup
+ * so add a test here to ensure that when one is provided
+ * libcrypo returns an error
+ */
+static int evp_test_name_parsing(void)
+{
+    EVP_MD *md;
+
+    if (TEST_ptr(md = EVP_MD_fetch(NULL, "SHA256:BogusName", NULL))) {
```

Comment:
Please use mainctx as the library context here.

Reply:
why? There are sevearal other tests in this file that follow the same pattern as above.  See test_alternative_default, and test_provider_unload_effective

Reply:
Both of them actually _depend_ on the NULL libctx being the one with just the null provider loaded. However this test should work even with any providers loaded.

Also I've noticed the condition should actually be `!TEST_ptr_null(md = EVP_MD_fetch(mainctx, "SHA256:BogusName", NULL))`

Reply:
It shouldn't be a negative test.  The test expects that, when an invalid name is passed to EVP_MD_fetch, that a null pointer is returned.  So in this test gett a valid pointer on the invalid name "SHA256:BogusName" is a failing test case

Reply:
changed to use of mainctx